index.js

```diff
@@ -1,166 +1,23 @@
 #!/usr/bin/env node
 
-const { request, stream, setGlobalDispatcher, Agent } = require('undici')
-const EE = require('events')
-const fs = require('fs')
-const path = require('path')
-const os = require('os')
-const debug = require('debug')('is-my-node-vulnerable')
-const satisfies = require('semver/functions/satisfies')
-const { danger, vulnerableWarning, bold, separator, allGood } = require('./ascii')
-const nv = require('@pkgjs/nv')
-
-setGlobalDispatcher(new Agent({ connections: 20 }))
-
-const CORE_RAW_URL = 'https://raw.githubusercontent.com/nodejs/security-wg/main/vuln/core/index.json'
-
-let lastETagValue
-
-const coreLocalFile = path.join(__dirname, 'core.json')
-const ETagFile = path.join(__dirname, '.etag')
-
-async function readLocal (file) {
-  return require(file)
-}
-
-function loadETag () {
-  if (fs.existsSync(ETagFile)) {
-    debug('Loading local ETag')
-    lastETagValue = fs.readFileSync(ETagFile).toString()
-  }
-}
-
-function updateLastETag (etag) {
-  lastETagValue = etag
-  fs.writeFileSync(ETagFile, lastETagValue)
-}
-
-async function fetchCoreIndex () {
-  const abortRequest = new EE()
-  await stream(CORE_RAW_URL, { signal: abortRequest }, ({ statusCode }) => {
-    if (statusCode !== 200) {
-      console.error('Request to Github failed. Aborting...')
-      abortRequest.emit('abort')
-      process.nextTick(() => { process.exit(1) })
-    }
-    return fs.createWriteStream(coreLocalFile, { flags: 'w', autoClose: true })
-  })
-  return readLocal(coreLocalFile)
-}
-
-async function getCoreIndex () {
-  const { headers } = await request(CORE_RAW_URL, { method: 'HEAD' })
-  if (!lastETagValue || lastETagValue !== headers.etag || !fs.existsSync(coreLocalFile)) {
-    updateLastETag(headers.etag)
-    debug('Creating local core.json')
-    return fetchCoreIndex()
+const isOldEnough = require('./eol-versions')
+const ascii = require('./ascii')
+
+// To guarantee support on older versions and do not drastically impact
+// the maintenance of this module, we check if process.version is too old
+// and throw EOL warning when true.
+if (isOldEnough(process.version)) {
+  console.log(ascii.danger)
+  const msg = process.version + ' is end-of-life. There are high chances of being vulnerable. Please upgrade it.'
+  console.log(msg)
+  process.exit(1)
+} else {
+  // // CLI
```

Comment:
nit: extra forward slashes
```suggestion
  // CLI
```